Source/WebInspectorUI/UserInterface/Views/Layers3DContentView.js

```diff
@@ -284,11 +296,88 @@ WI.Layers3DContentView = class Layers3DContentView extends WI.ContentView
     _createLayerGroup(layer) {
         let layerGroup = new THREE.Group;
         layerGroup.userData.layer = layer;
-        layerGroup.add(this._createLayerMesh(layer.bounds), this._createLayerMesh(layer.compositedBounds, true));
+
+        let fillMesh = this._createLayerMesh(layer.bounds, false, null);
+        let outlineMesh = this._createLayerMesh(layer.compositedBounds, true, null);
+        layerGroup.add(fillMesh, outlineMesh);
+
+        if (this._showLayerContent) {
+            this._loadLayerTexture(layer, (texture) => {
+                if (!texture)
+                    return;
+
+                if (!layerGroup.parent) {
+                    texture.dispose();
+                    return;
+                }
+
+                layerGroup.remove(fillMesh);
+                if (fillMesh.geometry)
+                    fillMesh.geometry.dispose();
+                if (fillMesh.material) {
+                    if (fillMesh.material.map)
+                        fillMesh.material.map.dispose();
+                    fillMesh.material.dispose();
+                }
+
+                let texturedMesh = this._createLayerMesh(layer.bounds, false, texture);
+                layerGroup.add(texturedMesh);
+
+                layerGroup.remove(outlineMesh);
+                layerGroup.add(outlineMesh);
+            });
+        }
+
         return layerGroup;
     }
 
-    _createLayerMesh({x, y, width, height}, isOutline = false)
+    _loadLayerTexture(layer, callback) {
+        if (this._pendingTextureLoads.has(layer.layerId))
+            this._pendingTextureLoads.get(layer.layerId).cancelled = true;
+
+        let loadContext = { cancelled: false };
```

Comment:
it seems like the only reason we're using an object here is so we still have something to reference even if it's removed from `_pendingTextureLoads` (i.e. there's no other properties on this object other than `cancelled`).  this definitely doesn't feel like the most ideal approach, but im not sure there would be anything more efficient (other than maybe an incrementing "generation"-style value but that has it's own issues) so i suppose this is fine for now